query-algebrizer/src/clauses/not.rs

```diff
@@ -56,3 +55,146 @@ impl ConjoiningClauses {
+        let query = r#"
+            [:find ?x
+             :where [?x :foo/knows "John"]
+                    (not [?x :foo/parent "Ámbar"]
+                         [?x :foo/knows "Daphne"])]"#;
+        let cc = alg(&schema, query);
+
+        let vx = Variable::from_valid_name("?x");
+        let d0 = "datoms00".to_string();
+        let d1 = "datoms01".to_string();
+        let d2 = "datoms02".to_string();
+        let d0e = QualifiedAlias::new(d0.clone(), DatomsColumn::Entity);
+        let d0a = QualifiedAlias::new(d0.clone(), DatomsColumn::Attribute);
+        let d0v = QualifiedAlias::new(d0.clone(), DatomsColumn::Value);
+        let d1e = QualifiedAlias::new(d1.clone(), DatomsColumn::Entity);
+        let d1a = QualifiedAlias::new(d1.clone(), DatomsColumn::Attribute);
+        let d1v = QualifiedAlias::new(d1.clone(), DatomsColumn::Value);
+        let d2e = QualifiedAlias::new(d2.clone(), DatomsColumn::Entity);
+        let d2a = QualifiedAlias::new(d2.clone(), DatomsColumn::Attribute);
+        let d2v = QualifiedAlias::new(d2.clone(), DatomsColumn::Value);
+
+        let knows = QueryValue::Entid(66);
+        let subquery_knows = QueryValue::Entid(66);
+        let parent = QueryValue::Entid(67);
+        let john = QueryValue::TypedValue(TypedValue::typed_string("John"));
+        let ambar = QueryValue::TypedValue(TypedValue::typed_string("Ámbar"));
+        let daphne = QueryValue::TypedValue(TypedValue::typed_string("Daphne"));
+
+        let mut subquery = ConjoiningClauses::default();
+        subquery.from.append(&mut vec![SourceAlias(DatomsTable::Datoms, d1),
+                                  SourceAlias(DatomsTable::Datoms, d2)]);
+        subquery.column_bindings.insert(vx.clone(), vec![d1e.clone(), d2e.clone()]);
+        subquery.wheres.add(ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d1a.clone(), parent)));
+        subquery.wheres.add(ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d1v.clone(), ambar)));
+        subquery.wheres.add(ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d2a.clone(), subquery_knows)));
+        subquery.wheres.add(ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d2v.clone(), daphne)));
+        subquery.wheres.add(ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d1e.clone(), QueryValue::Column(d2e.clone()))));
+
+        let mut s = HashSet::with_capacity(1);
+        s.insert(ValueType::Ref);
+        subquery.known_types.insert(vx.clone(), s);
+
+        assert!(!cc.is_known_empty());
+        assert_eq!(cc.wheres, ColumnIntersection(vec![
+                ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d0a.clone(), knows.clone())),
+                ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d0v.clone(), john)),
+                ColumnConstraintOrAlternation::Constraint(ColumnConstraint::NotExists(ComputedTable::Subquery{ clauses: subquery })),
```

Comment:
I suspect `ComputedTable::Subquery` should change from being a "struct type" to a "tuple type", 'cos the name is confusing ("clauses") and it only has one field. That is, rather than

```
    Subquery: { clauses: ConjoiningClauses },
```

you'd have

```
    Subquery(ConjoiningClauses),
```

and use it that way here:

```
ColumnConstraint::NotExists(ComputedTable::Subquery(subquery))
```